docs/DeveloperGuide.md

```diff
@@ -276,12 +276,33 @@ The following activity diagram summarizes what happens when a user executes the
 
 #### Implementation
 
-To be added: Reminder implementation
+The remind mechanism is facilitated by `RemindCommand` and `RemindCommandParser`. It allows users to set a reminder message and date, for a contact from the contact list.
 
 The `Reminder` objects for each `Person` is stored in the `Person` object as a `SortedList` where the predicate for sorting is based on the `date` attribute in `Reminder`.
 
 The `UI` component then displays all the `Reminder` for all `Person` objects in `AddressBook.persons` in the `ReminderListPanel`.
 
+#### Example Usage
+
+Step 1: The user inputs `remind 1 r/zoom call d/2023-10-10` to create a reminder for the 2nd person in the displayed contact list.
+
+Step 2: `LogicManager` calls `AddressBookParser` with the user input.
+
+Step 3: `AddressBookParser` will parse the command word and create a new `ReminderCommandParser` and call its function `parse` with the rest of the user input as the arguments.
+
+Step 4: The `ReminderCommandParser#parse` will then parse the index, task and date to create a new `ReminderCommand` object.
+
+Step 5: The `LogicManager` then calls `ReminderCommand#execute`.
+
+Step 6: The `ReminderCommand` calls `Model#getFilteredPersonList` to get the filtered `List` of `Person` objects.
+
+Step 7: The `ReminderCommand` gets the `Person` to be edited from the `List`
+
+Step 8: The `ReminderCommand` creates a new edited `Person`, with the given `Reminder`
+
+Step 10: `ReminderCommand` then returns a new `CommandResult` with the result of the execution.
+
+![Sequence diagram for the Remind Command](images/ReminderSequenceDiagram.png)
```

Comment:
I think you can add the limitation of PlantUML that causes the lifeline for ReminderCommandParser to continue even after the destroy marker (X).

Reply:
Thanks! Will update 🤩